src/jenkins_job_insight/diagnostic_archive.py

```
@@ -0,0 +1,476 @@
+                    f"Tar data_filter rejected entries, falling back to manual filtering: {exc}"
+                )
+
+        # Manual member-by-member extraction with filtering
+        safe_members = []
+        for member in tar.getmembers():
+            if member.issym() or member.islnk():
+                link_target = Path(extract_dir / member.linkname).resolve()
+                if not str(link_target).startswith(str(extract_dir.resolve())):
+                    logger.warning(
+                        f"Skipping tar entry with symlink outside extraction dir: "
+                        f"{member.name} -> {member.linkname}"
```

Ruff (0.15.5)

[warning] Logging statement uses f-string

(G004)